File: Pesquisa e Ordenação/Linguagens/Python/CompareToo/aluno.py
class Aluno:
    def __init__(self, nome, idade):
        self.nome = nome
        self.idade = idade

    def __str__(self):
        return f"Aluno{{nome={self.nome}, idade={self.idade}}}"

    def __eq__(self, other):
        if not isinstance(other, Aluno):
            return False
        return self.nome == other.nome and self.idade == other.idade

    def __lt__(self, other):
        if self.nome != other.nome:
            return self.nome < other.nome
        return self.idade < other.idade

File: Pesquisa e Ordenação/Linguagens/Java/CompareTo/Aluno.java
package CompareTo.Java;

import java.util.Objects;

public class Aluno implements Comparable<Aluno> {
    String nome;
    int idade;

    public Aluno(String nome, int idade) {
        this.nome = nome;
        this.idade = idade;
    }
    
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    @Override
    public String toString() {
        return "Aluno{" + "nome=" + nome + ", idade=" + idade + '}';
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Aluno other = (Aluno) obj;
        if (this.idade != other.idade) {
            return false;
        }
        return Objects.equals(this.nome, other.nome);
    }

    @Override
    public int compareTo(Aluno o) {
        if (this.nome.compareTo(o.nome) != 0) {
            return this.nome.compareTo(o.nome);
        }
        return this.idade - o.idade;
    }
   
}
